### PR TITLE
feat(tags): Allows tags endpoint to optionally return human readable values

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds a `readable` option to the `/tags` endpoint that includes more human readable values for certain tag values.
For now, this is only supported by the `device` tag which converts device model numbers to their more readable marketing names.

### CODE REVIEW

File: src/sentry/api/endpoints/group_tags.py [hunk 1]
```diff
@@ -37,4 +38,20 @@ def get(self, request: Request, group: Group) -> Response:
...
+        show_readable_tag_values = request.GET.get("readable")
+        if show_readable_tag_values:
+            add_readable_tag_values(data)
+
+        return Response(data)
+
+
+def add_readable_tag_values(data: Any) -> Any:
+    # Map device tag to a more readable value if possible
+    device_tag = next((tag for tag in data if tag["key"] == "device"), None)
```
Comment (wmak) [inline src/sentry/api/endpoints/group_tags.py:52]: nit: Could this just be a for loop with a break instead?

File: src/sentry/api/endpoints/group_tags.py [hunk 2]
```diff
@@ -37,4 +38,20 @@ def get(self, request: Request, group: Group) -> Response:
...
+        data = serialize(tag_keys, request.user)
+
+        show_readable_tag_values = request.GET.get("readable")
+        if show_readable_tag_values:
+            add_readable_tag_values(data)
+
+        return Response(data)
+
+
+def add_readable_tag_values(data: Any) -> Any:
```
Comment (wmak) [inline src/sentry/api/endpoints/group_tags.py:50]: Should the return type be None?

File: src/sentry/api/helpers/mobile.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from sentry.api.helpers.android_models import ANDROID_MODELS
```
Comment (wmak) [inline src/sentry/api/helpers/mobile.py:1]: With how giant this file is, do we know if there's any performance impact? Like I wonder if this is faster as a giant python file, or a CSV that we read in

**Review by wmak — COMMENTED** [review]
lgtm, but the 18k file sorta worries me, could we get some performance numbers using the timeit module (or sentry)

File: src/sentry/api/helpers/mobile.py (hunk 3 above)
Comment (edwardgou-sentry) [reply src/sentry/api/helpers/mobile.py:1]: So I tested in pytest by wrapping the request in `timeit` and took the average of 500 `/tags` responses.
On this branch with the new `readable` flag, the average was 0.0835806546400001s
On this branch without the `readable` flag, the average was 0.08407600666000004s
And on `master` without any changes, the average was 0.08446629567999978s

Seems like there isn't any noticable performance impact? The difference between all test cases seem small enough that they could just be random.

File: src/sentry/api/endpoints/group_tags.py (hunk 2 above)
Comment (edwardgou-sentry) [reply src/sentry/api/endpoints/group_tags.py]: yup 😄

File: src/sentry/api/endpoints/group_tags.py (hunk 1 above)
Comment (edwardgou-sentry) [reply src/sentry/api/endpoints/group_tags.py]: Updated to a for loop!

**Review by wmak — APPROVED** [review]
Can we include a comment in `src/sentry/api/helpers/android_models.py` explaining how to update it? (ie. step 1 download file from ___, step 2 .... etc.)

**Review by wedamija — COMMENTED** [review]
I feel like I've seen this mapping (or a similar mapping) before, maybe related to profiling? Are we sure we're not duplicating something that already exists?

**Comment by edwardgou-sentry** [comment]
> I feel like I've seen this mapping (or a similar mapping) before, maybe related to profiling? Are we sure we're not duplicating something that already exists?

I think there's been a link to some external mapping being shared on slack but none of that has been committed to the sentry repo yet from what I've been told.
Also did a quick search for some of the values in the mapping in the sentry repo and didnt find any results so I think this would be the first time we introduce a mapping like this?

**Pull request merged**